src/util/std_expr.h

```diff
@@ -4654,6 +4657,23 @@ class class_method_descriptor_exprt : public nullary_exprt
   }
 };
 
+inline void validate_expr(const class class_method_descriptor_exprt &value)
+{
+  validate_operands(value, 0, "class method descriptor must not have operands");
+  DATA_INVARIANT(
+    !value.mangled_method_name().empty(),
+    "class method descriptor must have a mangled method name.");
+  DATA_INVARIANT(
+    !value.class_id().empty(), "class method descriptor must have a class id.");
+  DATA_INVARIANT(
+    !value.base_method_name().empty(),
+    "class method descriptor must have a base method name.");
+  DATA_INVARIANT(
```

Comment:
:bulb: Rather than enforcing this data invariant, why not have `get_identifier` return `id2string(value.class_id()) + "." + id2string(value.mangled_method_name())`?

Reply:
I considered doing that. However I didn't wish to change the underlying data structure (yet), just in case there are any remaining direct accesses. I think ideally we would also keep the mangled name as a separate `method_base_name` and `method_descriptor`, so that the `mangled_method_name` would return `id2string(base_method_name()) + ":" + id2string(method_descriptor())`.